### PR TITLE
Add shares to CDP

### PR DESCRIPTION
**Pull request by CoreyMartin**

### CODE REVIEW

File: contracts/CDPosition.sol [hunk 1]
```diff
@@ -121,7 +135,7 @@ contract CDPosition {
         return _nftCDP[nftID].lvUSDBorrowed;
     }
 
-    function getFirstCycle(uint256 nftID) external view nftIDMustExist(nftID) returns (bool) {
-        return _nftCDP[nftID].firstCycle;
+    function getShares(uint256 nftID) external view nftIDMustExist(nftID) returns (uint256) {
+        return _nftCDP[nftID].shares;
```
Comment (yotamdaniel) [inline contracts/CDPosition.sol:139]: Might want to add nftIDMustExist(nftId)

File: contracts/Coordinator.sol [hunk 2]
```diff
@@ -70,10 +70,11 @@ contract Coordinator is ICoordinator {
         address sharesOwner
     ) external override {
         /// Transfer collateral to vault, mint shares to shares owner
-        VaultOUSD(_tokenVaultOUSD).deposit(amount, sharesOwner);
+        uint256 shares = VaultOUSD(_tokenVaultOUSD).deposit(amount, sharesOwner);
+        console.log("shares", shares);
```
Comment (yotamdaniel) [inline contracts/Coordinator.sol:74]: remove log  + import

File: test/CDPosition.js [hunk 3]
```diff
@@ -66,6 +68,34 @@ describe("CDPosition test suit", async function () {
...
+            // validate CDP Values
+            await validateCDP(NFT_ID, BASIC_OUSD_PRINCIPLE, 0, BASIC_OUSD_PRINCIPLE, 0, 0);
+            // create a new position
+            await cdp.addSharesToPosition(NFT_ID, 20);
+            await validateCDP(NFT_ID, BASIC_OUSD_PRINCIPLE, 0, BASIC_OUSD_PRINCIPLE, 0, 20);
+        });
+
+        it("Should remove shares from position", async function () {
+            // create a new position
+            await cdp.addSharesToPosition(NFT_ID, 20);
```
Comment (yotamdaniel) [inline test/CDPosition.js:84]: Can still pass expect even if both of those fail. Maybe remove a different number of shares and expect the diff

File: test/Coordinator.js [hunk 4]
```diff
@@ -5,37 +5,45 @@ const { ContractTestContext } = require("./ContractTestContext");
...
     before(async function () {
         mainnetHelper.helperResetNetwork(mainnetHelper.defaultBlockNumber);
 
         r = new ContractTestContext();
         await r.setup();
 
-        endUserSigner = r.addr1;
+        // Object under test
         coordinator = r.coordinator;
-        sharesOwnerAddress = coordinator.address;
```
Comment (yotamdaniel) [inline test/Coordinator.js:20]: why move this and the above?

File: contracts/Coordinator.sol [hunk 5]
```diff
@@ -70,10 +70,10 @@ contract Coordinator is ICoordinator {
         address sharesOwner
     ) external override {
         /// Transfer collateral to vault, mint shares to shares owner
-        VaultOUSD(_tokenVaultOUSD).deposit(amount, sharesOwner);
+        uint256 shares = VaultOUSD(_tokenVaultOUSD).deposit(amount, sharesOwner);
         // create CDP position with collateral
-        // TODO : !!!!!! update shares allocation to position !!!!!!!
         CDPosition(_tokenCDP).createPosition(nftId, amount);
+        CDPosition(_tokenCDP).addSharesToPosition(nftId, shares);
```
Comment (yotamdaniel) [inline contracts/Coordinator.sol:76]: perfect thanks!

File: test/CDPosition.js (hunk 3 above)
Comment (CoreyMartin) [reply test/CDPosition.js]: fixed!

File: test/Coordinator.js (hunk 4 above)
Comment (CoreyMartin) [reply test/Coordinator.js:20]: reverted

**Review by yotamdaniel — APPROVED** [review]

**Pull request merged**